### PR TITLE
Ark Integration for Archival

### PR DESCRIPTION
**Pull request by dgoodwin**
This is the first round of support for integrating with Ark to handle our archival/unarchival needs. An Ark backup resource will now be created in etcd when we detect a namespace should be archived due to inactivity or cluster capacity. Ark handles the storage in S3 and PV snapshotting. We monitor for when the backup completes and optionally delete the namespace once successful. This is disabled by default and may not be used in production, we expect the user registration app to handle namespace deletion.

Our homemade archival code is now removed as is the integration test framework, we have no integration tests for this portion of the code given the complexity of deploying and interacting between Ark and Archivist. Instead we have a strong suite of unit tests to ensure we accurate calculate when projects should be archived, and submit a sane Backup resource to Ark.

First commit is separated to isolate vendor updates in git history.

PR also introduces use of OpenShift's new client-go library. This cuts our deps from 330mb to 100mb and dramatically improves compile and test time.

### CODE REVIEW

File: pkg/clustermonitor/clustermonitor.go [hunk 1]
```diff
@@ -143,17 +189,22 @@ func (a *ClusterMonitor) Run() {
...
 	duration := a.cfg.MonitorCheckInterval
 	tickerTime := time.Duration(duration)
 	ticker := time.NewTicker(tickerTime)
 
 	go func() {
-		for t := range ticker.C {
-			log.Info("Checking capacity at: ", t)
-			go a.checkCapacity()
+		for range ticker.C {
+			log.Info("checking cluster capacity")
```
Comment (markturansky) [inline pkg/clustermonitor/clustermonitor.go]: perhaps a log level to filter this noise from the logs? I'm sure it's useful for development but becomes clutter in a log.

Comment (dgoodwin) [reply pkg/clustermonitor/clustermonitor.go]: Goal was debug for chatty stuff, info for high level overview of a capacity check as well as notification that we actually archived someone, and warn for anything troubling but we're still ok. I'll take another pass through and lower this one in particular.

Comment (markturansky) [reply pkg/clustermonitor/clustermonitor.go]: I think the appropriate level of logging is what you would actually need/want to debug something happening in prod. With that frame of mind, generic logging messages become spam, especially if they are repeated a lot.

One can argue a well behaved program should say nothing :)

Comment (dgoodwin) [reply pkg/clustermonitor/clustermonitor.go]: What you're describing would be achieved by setting to warn, if you trust the code enough and don't care what's going on in general, just log if something's going wrong or looks off.

Nothing at info should be noisy, this loop would execute about once or twice a day, and info is intended to be concise info I need to see for how it's behaving in prod dry-run.  (debug would be too noisy and I don't intend to use beyond development or serious debugging) We need a level that's a concise summary of what's happening so we can see how this is playing out in real world.

**Comment by dgoodwin** [comment]
Reduced a couple log statements to debug, here's what the output looks like now for a loop, which would run maybe 2-4 times a day:

```
INFO[0000] calculating namespaces to be archived         checkTime="2017-11-20 09:15:23.028995419 -0400 AST" highWatermark=8 lowWatermark=6 maxInactive="2017-11-19 09:15:23.028995419 -0400 AST" minInactive="2017-11-20 09:14:23.028995419 -0400 AST"
DEBU[0000] skipping protected namespace                  namespace=kube-system
DEBU[0000] skipping protected namespace                  namespace=openshift-infra
DEBU[0000] skipping protected namespace                  namespace=openshift-node
DEBU[0000] skipping protected namespace                  namespace=default
DEBU[0000] skipping protected namespace                  namespace=heptio-ark
DEBU[0000] skipping protected namespace                  namespace=kube-public
DEBU[0000] skipping protected namespace                  namespace=logging
DEBU[0000] skipping protected namespace                  namespace=openshift
DEBU[0000] calculating last activity time                builds=0 namespace=testnamespace rcs=0
DEBU[0000] no builds or RCs for namespace, using project creation time for last activity  lastActivity="2017-11-01 09:20:02 -0300 ADT" namespace=testnamespace
DEBU[0000] calculated last activity                      lastActivity="2017-11-01 09:20:02 -0300 ADT" namespace=testnamespace
INFO[0000] found namespace over max inactive time        checkTime="2017-11-20 09:15:23.028995419 -0400 AST" lastActivity="2017-11-01 09:20:02 -0300 ADT" maxInactive="2017-11-19 09:15:23.028995419 -0400 AST" namespace=testnamespace
INFO[0000] last activity totals                          somewhatInactive=0 totalNamespaces=9 veryInactive=1
DEBU[0000] looking for 2 semi-inactive namespaces to archive 
INFO[0000] found 1 namespaces to archive                
WARN[0000] unable to reach namespace capacity low watermark  lowWatermark=6 newNSCount=8
INFO[0000] archiving namespace                           namespace=testnamespace
DEBU[0000] creating backup in ark                        name=testnamespace-20171120091523
ERRO[0000] error archiving namespace: skipping archival, no openshift.io/requester annotation found on project: testnamespace 
```

For INFO we'd be talking about roughly 5 lines per loop aside, and then 2 lines per project that needs to get archived.

**Pull request merged**